crates/sui-source-validation/src/lib.rs

```diff
@@ -221,90 +215,134 @@ impl<'a> BytecodeSourceVerifier<'a> {
             ),
         }
     }
-}
-
-fn get_module_bytes_map(
-    compiled_package: &CompiledPackage,
-    include_deps: bool,
-    source_mode: SourceMode,
-) -> Result<ModuleBytesMap, SourceVerificationError> {
-    let mut map: ModuleBytesMap = HashMap::new();
-
-    #[allow(clippy::type_complexity)]
-    fn make_map_entry(
-        package: &Symbol,
-        named_compiled_module: &NamedCompiledModule,
-        subst_addr: Option<AccountAddress>,
-    ) -> Result<((AccountAddress, Symbol), (Symbol, Vec<u8>)), SourceVerificationError> {
-        let module = named_compiled_module.name;
-        let address = subst_addr.unwrap_or_else(|| named_compiled_module.address.into_inner());
-        let mut bytes = vec![];
-
-        // Replace the zero address entries in the module if needed
-        if let Some(new_address) = subst_addr {
-            let mut compiled_module = named_compiled_module.module.clone();
-            let self_handle = compiled_module.self_handle().clone();
-            let self_address_idx = self_handle.address;
-
-            let addrs = &mut compiled_module.address_identifiers;
-            let Some(address_mut) = addrs.get_mut(self_address_idx.0 as usize) else {
-                let name = compiled_module.identifier_at(self_handle.name);
-                return Err(SourceVerificationError::InvalidModuleFailure {
-                    name: name.to_string(),
-                    message: "Self address field missing".to_string()
-                });
-            };
 
-            if *address_mut != AccountAddress::ZERO {
-                let name = compiled_module.identifier_at(self_handle.name);
-                return Err(SourceVerificationError::InvalidModuleFailure {
-                    name: name.to_string(),
-                    message: "Self address already populated".to_string(),
-                });
-            };
+    async fn on_chain_bytes(
+        &self,
+        addresses: impl Iterator<Item = AccountAddress>,
+    ) -> Result<OnChainBytes, SourceVerificationError> {
+        let mut map = OnChainBytes::new();
+
+        for addr in addresses {
+            let SuiRawMovePackage { module_map, .. } = self.pkg_for_address(&addr).await?;
```

Comment:
I wanted to be able to make these requests in parallel, but the fact that `tokio::spawn` requires a `'static` lifetime bound which `self` will not outlive was a source of issues.

Any tips appreciated!

Reply:
Summoning @bmwill

Reply:
I just pushed a commit on top of this PR's branch that changes these to be done concurrently (instead of 1 after another). Feel free to take it, change it, or remove it.

Reply:
Love it, thank you! Also, TIL that you can push to other people's forks. @bmwill, operationally, how does that work -- are you just setting my fork as an upstream and pushing to it?

Reply:
I had to tweak the use of `futures` to just `join_all` in the end because the streams API was inducing a higher-kinded lifetime error in another part of the codebase, strangely, but `join_all` works!  Thanks for the pointers.

Reply:
There's a setting you can toggle to "allow maintainers to modify" that only gives permissions to maintainers of an upstream repo to push to the branch you use for your PR on your own personal fork